service/history/api/respondworkflowtaskcompleted/api.go

```diff
@@ -195,6 +195,13 @@ func (handler *WorkflowTaskCompletedHandler) Invoke(
 		return nil, serviceerror.NewNotFound("Workflow task not found.")
 	}
 
+	// We don't accept the request to create a new workflow task if the workflow is paused.
```

Comment:
Still seems weird/inconsistent to me that if workflow task is not heartbeating, we accept the result, otherwise we drop the response and let it timeout. But I understand this is what's been agreed on.